academy/4-my-own-chain/server-side.md

```diff
@@ -0,0 +1,802 @@
+    }
+
+    const init = async () => {
+        setTimeout(poll, 1)
+    }
+
+    const poll = async () => {
+        console.log(new Date(Date.now()).toISOString(), "TODO poll")
+        timer = setTimeout(poll, pollIntervalMs)
+    }
+
+    process.on("SIGINT", () => {
+        if (timer) clearTimeout(timer)
+        saveDb()
+            .then(() => {
+                console.log(`${dbFile} saved`)
+            })
+            .catch(console.error)
+            .finally(() => {
+                server.close(() => {
+                    console.log("server closed")
+                    process.exit(0)
+                })
+            })
+    })
+
+    const server: Server = app.listen(port, () => {
+        init()
+            .catch(console.error)
+            .then(() => {
+                console.log(`\nserver started at http://localhost:${port}`)
+            })
+    })
+}
+```
+
+</ExpansionPanel>
+
+Notice how:
+
+1. The timer is set at the end of the previous poll, just in case the indexing takes longer than the interval.
+2. The _database_ is purely in memory as it runs and is saved on exit by catching the interruption signal.
+
+Create an `index.ts` that describe how to run it:
+
+```typescript
+require("./indexer").createIndexer().then(console.log).catch(console.error)
+
+```
+
+And in `package.json`, add a `run` target:
+
+```json
+"scripts": {
+    ...
+    "dev": "npx ts-node index.ts"
+}
+```
+
+Not to forget the _database_, `db.json`:
+
+```json
+{
+    "status": {
+        "block": {
+            "height": -1
+        }
+    },
+    "players": {},
+    "games": {}
+}
+```
+
+Make sure it works:
+
+```sh
+$ npm run dev
+```
+
+It should print:
+
+```
+> checkers-server@1.0.0 dev
+> npx ts-node index.ts
+
+server started at http://localhost:3001
+```
+
+Now you can test the endpoints. Omit the `| jq` beautifier if it is not installed on your system:
+
+<CodeGroup>
+<CodeGroupItem title="status" active>
+
+```sh
+$ curl localhost:3001/status | jq
+```
+
+It should return:
+
+```json
+{
+  "block": {
+    "height": -1
+  }
+}
+```
+
+</CodeGroupItem>
+<CodeGroupItem title="player info">
+
+```sh
+$ curl localhost:3001/players/cosmos123 | jq
+```
+
+It should return:
+
+```json
+{
+  "gameCount": 0,
+  "gameIds": []
+}
+```
+
+</CodeGroupItem>
+<CodeGroupItem title="player games">
+
+```sh
+$ curl localhost:3001/players/cosmos123/gameIds | jq
+```
+
+It should return:
+
+```json
+[]
+```
+
+</CodeGroupItem>
+<CodeGroupItem title="game update">
+
+```sh
+$ curl -X PATCH localhost:3001/games/445 | jq
+```
+
+It should return:
+
+```json
+{
+  "result": "thank you"
+}
+```
+
+</CodeGroupItem>
+</CodeGroup>
+
+---
+
+## Add CosmJs `StargateClient`
+
+In order to connect to your Checkers blockchain, you need to create a client. Yhe client only needs read-only functionality because this server does not intend to submit transactions. Also, add the CosmJs types, which will assist you with blocks, transactions and events:
```

Comment:
```suggestion
In order to connect to your Checkers blockchain, you need to create a client. The client only needs read-only functionality because this server does not intend to submit transactions. Also, add the CosmJs types, which will assist you with blocks, transactions and events:
```